### DISCUSSION TITLE
Running Dockerless

### DISCUSSION THREAD

**Discussion by techfoxactual** [dec0dOS/zero-ui#30] [General]
Hiya,

How would I go about running this outside of docker? I have my ZT controller and networks setup etc already running outside docker, not real keen on introducing docker needlessly here, I can see that `yarn build` builds a deployment ready version of the project, however that's as far as I've got 😅

How can I deploy the built project? (It tells me it's made a build folder), ultimately I'll be putting this behind nginx.

Thanks!!!

**Comment by dec0dOS** [comment]
Hello, @knightian, you could always run your controller outside the docker, please refer to `Controller setup tips` in `README.md`.
ZeroUI is recommended to be running in Docker to provide seamless updates and ensure that all the users have the correct installation.

You could always proxy your Docker-based ZeroUI installation behind the nginx.

**Reply by techfoxactual** [reply]
@dec0dOS thanks yea I could, however I’d like to avoid the overhead of docker if possible. Given that `yarn dev` runs it in dev mode, surely there is a similar command I can run to run it in production?

**Reply by dec0dOS** [reply]
All the commands that are used for production deployment could be found in the Dockerfile: https://github.com/dec0dOS/zero-ui/blob/main/docker/zero-ui/Dockerfile

**Comment by techfoxactual** [comment]
Wonderful thanks! Yea I was stupidly trying to use docker-compose to work it out haha!

**Comment by techfoxactual** [comment]
@dec0dOS 

So I build it like the docker file, when I try access it I see:

GET /app/ 200 6.562 ms - -
GET / 302 4.464 ms - 52
GET /app/ 304 1.078 ms - -

But it's a empty white screen I see nothing :(

**Reply by dec0dOS** [reply]
@knightian, check the environmental variables in the `docker-compose.yml` file.

You could find the detail explanation in the `README.md`.

**Reply by techfoxactual** [reply]
Yep I set them, I set the admin and password and the others I can leave as the defaults

I also set NODE_ENV to production

**Reply by dec0dOS** [reply]
You should set `ZU_SECURE_HEADERS` to `false` if you're not using the HTTPS.

**Reply by techfoxactual** [reply]
Ok, now I'm getting 404

GET /app/ 404 1.506 ms - 25
GET /app/ 404 0.802 ms - 25

**Reply by dec0dOS** [reply]
You have to debug it yourself, that could be anything. As I said before, a Docker-based installation is recommended. Check again the environmental variables, especially the `ZU_SERVE_FRONTEND`.

**Reply by techfoxactual** [reply]
In my .profile I have

export ZU_DEFAULT_USERNAME="xxxxxx"
export ZU_DEFAULT_PASSWORD="xxxxxxx"
export NODE_ENV=production
export ZU_SECURE_HEADERS=false
export ZU_SERVE_FRONTEND=true

**Reply by techfoxactual** [reply]
Oh it's working now, I see, I had to build it specifically at /app/

**Comment by techfoxactual** [comment]
@dec0dOS love ya work mate, have it running as a systemd service all good!!!
